src/Moryx.Communication.Mqtt/README.md

```diff
@@ -0,0 +1,265 @@
+- In your `program.cs` add the following:
+
+```cs
+ builder.Services.AddMqttClient() // must be present
+                 .AddMqttEndpoints(); // <-- this add supports for the endpoints
+```
+- Create a new C# class that implements the `IMqttEndpoint`:
+```cs
+ public class Blabla : IMqttEndpoint 
+ {
+    private IMyDependency _dependency;
+    public Blabla(IMyDependency dependency)
+    {
+        _dependency = dependency;
+    }
+
+    public void Map(IMqttRouteBuilder route)
+    {
+        //topic template is supported here
+        route.MapGet("my-topic/{id:int}/method/{methodName}", context =>
+        {
+             var messageBuilder = new MqttApplicationMessageBuilder();
+            messageBuilder.WithTopic("another-topic");
+            messageBuilder.WithPayload(MqttMessageSerialization.GetJsonPayload("Hello World!!!!"));
+            return messageBuilder.Build();
+        });
+    }
+}
+```
+
+There is a minimal version of the Mqtt endpoint, that you can use directly in the `program.cs`:
+
+```cs
+builder.Services.AddMqttClient()
+                .AddMqttEndpoints(route =>
+                {
+                    route.MapGet("my-topic/{id:int}/method/{methodName}", context =>
+                    {
+                        var messageBuilder = new MqttApplicationMessageBuilder();
+                        messageBuilder.WithTopic("another-topic");
+                        messageBuilder.WithPayload(MqttMessageSerialization.GetJsonPayload("Hello World!!!!"));
+                        return messageBuilder.Build();
+                    });
+                    // optional : this behave like an eventStream
+                    route.MapPost(stream =>
+                    {
+                        var messageBuilder = new MqttApplicationMessageBuilder();
+                        messageBuilder.WithTopic("another-topic");
+                        var timer = new Timer(async arg =>
+                        {
+                            var messageBuilder = new MqttApplicationMessageBuilder();
+                            messageBuilder.WithPayload(MqttMessageSerialization.GetJsonPayload("Hello World!!!!"));
+                            var message = messageBuilder.Build();
+                            await stream.WriteAsync(message);
+                        }, null, 0, 10 * 1000);
+                    });
+                });
+```
+This can be useful for quick prototyping and testing.
+
+### Available features in endpoints
+Endpoints topics support constraint just like in ASP.net controller route.
+#### Route constraint
+
+The available constraints are:
+
+- {name: int}
+- {name: int?}
+- {name: string}
+- {name: string?}
+- {name: decimal}
+- {name: decimal?}
+- {name: bool}
+- {name: bool?}
+- {name: datetime}
+- {name: datetime?}
+- {name: double}
+- {name: double?}
+- {name: float}
+- {name: float?}
+- {name: guid}
+- {name: guid?}
+- {name: long}
+- {name: long?}
+- {name?}
+- {name/*}
```

Comment:
Where do they come from? Maybe reference another doc instead of copy the stuff into these docs.